lbry/lbry/dht/node.py

```diff
@@ -245,3 +227,71 @@ def accumulate_peers(self, search_queue: asyncio.Queue,
                          asyncio.Queue, asyncio.Task]:
         q = peer_queue or asyncio.Queue(loop=self.loop)
         return q, self.loop.create_task(self._accumulate_search_junction(search_queue, q))
+
+    async def _maintain_connection(self, seed_peers: typing.List['KademliaPeer'], timeout: float = None):
+        elapsed = 0
+        while True:
+            if timeout and elapsed > timeout:
+                return
+            if not self.protocol.routing_table.get_peers():
+                start = self.loop.time()
+                if self.joined.is_set():
+                    self.joined.clear()
+                self.protocol.peer_manager.reset()
+                self.protocol.ping_queue.enqueue_maybe_ping(*seed_peers, delay=0.0)
+                seed_peers.extend(await self.peer_search(self.protocol.node_id, shortlist=seed_peers, count=32))
+                if self.protocol.routing_table.get_peers():
+                    elapsed = 0
+                else:
+                    elapsed += self.loop.time() - start
+                    continue
+            elif not self.joined.is_set():
+                self.joined.set()
+                log.info(
+                    "Joined DHT, %i peers known in %i buckets", len(self.protocol.routing_table.get_peers()),
+                    self.protocol.routing_table.buckets_with_contacts()
+                )
+            await asyncio.sleep(1, loop=self.loop)
+
+    async def _persist_routing_table(self):
+        peers = []
+        resolved_addresses = await self._resolve_addresses(self._known_node_urls)
```

Comment:
OK, but I thought this change was mainly to avoid pinging the seeds on startup?

Reply:
The purpose is to avoid relying on our seed nodes for startup, and also to speed up joining the network. The seeds shouldn't be relied upon or favored over viable alternate nodes, but they don't need to be avoided.